features/nanostack/FEATURE_NANOSTACK/targets/TARGET_SL_RAIL/NanostackRfPhyEfr32.cpp

```diff
@@ -152,6 +187,103 @@ static int8_t rf_start_cca(uint8_t *data_ptr, uint16_t data_length, uint8_t tx_h
+                rx_queue[rx_queue_tail] = NULL;
+
+                rx_queue_tail = (rx_queue_tail + 1) % SL_QUEUE_SIZE;
+            }
+
+            for (int i=0; i < SL_QUEUE_SIZE; i++) {
+                // Simplified qu
+                void* handle = (void*) rx_queue[i];
+                if (handle == NULL) {
+                    continue;
+                }
+
+                RAIL_RxPacketInfo_t* info = (RAIL_RxPacketInfo_t*) memoryPtrFromHandle(handle);
+
+                device_driver.phy_rx_cb(
+                        info->dataPtr + 1,
+                        info->dataLength - 1, 
+                        255, 
+                        info->appendedInfo.rssiLatch, 
+                        rf_radio_driver_id);
+
+                memoryFree(handle);
+                rx_queue[i] = NULL;
+            }
+        } else if (event.value.signals & SL_TX_DONE) {
+            tr_debug("rf_thread_loop: SL_TX_DONE signal received\n");
+            device_driver.phy_tx_done_cb(rf_radio_driver_id,
+                    current_tx_handle,
+                    PHY_LINK_TX_SUCCESS,
+                    1,
+                    1);
+        } else if (event.value.signals & SL_ACK_RECV) {
+            tr_debug("rf_thread_loop: SL_ACK_RECV signal received\n");
+            device_driver.phy_tx_done_cb( rf_radio_driver_id,
+                    current_tx_handle,
+                    (event.value.signals & SL_ACK_PEND) ? PHY_LINK_TX_DONE_PENDING : PHY_LINK_TX_DONE,
+                    1,
+                    1);
+        } else if (event.value.signals & SL_ACK_TIMEOUT) {
+            tr_debug("rf_thread_loop: SL_ACK_TIMEOUT signal received\n");
+            waiting_for_ack = false;
+            device_driver.phy_tx_done_cb(rf_radio_driver_id,
+                    current_tx_handle,
+                    PHY_LINK_TX_FAIL,
+                    1,
+                    1);
+        } else if(event.value.signals & SL_TX_ERR) {
+            tr_debug("rf_thread_loop: SL_TX_ERR signal received\n");
+            device_driver.phy_tx_done_cb( rf_radio_driver_id,
+                    current_tx_handle,
+                    PHY_LINK_CCA_FAIL,
+                    8,
+                    1);
+        } else if(event.value.signals & SL_CAL_REQ) {
+            tr_debug("rf_thread_loop: SL_CAL_REQ signal received (unhandled)\n");
+        } else if(event.value.signals & SL_RXFIFO_ERR) {
+            tr_debug("rf_thread_loop: SL_RXFIFO_ERR signal received (unhandled)\n");
+        } else if(event.value.signals & SL_TXFIFO_ERR) {
+            tr_debug("rf_thread_loop: SL_TXFIFO_ERR signal received (unhandled)\n");
+        } else if(event.value.signals & SL_QUEUE_FULL) {
+            tr_debug("rf_thread_loop: SL_QUEUE_FULL signal received (packet dropped)\n");
+        } else {
+            tr_debug("rf_thread_loop unhandled event status: %d value: %d\n", event.status, event.value.signals);
+        }
+
+        platform_exit_critical();
+    }
+}
```

Comment:
Overall, I would recommend to remove all debugging prints from this function unless it actually means error, or unhandled signal.